cjwkernel/thrift.thrift

```diff
@@ -203,6 +203,14 @@ struct I18nMessage {
    * For instance, `{"nColumns": 3, "exampleColumn": "Column X"}`
    */
   2: map<string, I18nArgument> arguments
+  
+  /**
+   * An indication of where the message is coming from.
+   * - An empty map means it's coming from workbench itself
+   * - A map with key `"module_id"` means it's coming from a module
```

Comment:
Thrift is strongly-typed. This should be a `3: I18nMessageSource source`: a union of a `module_id` str or a `library` str. See `ParamValue` for an example of declaring union values. (Basically, a union looks remarkably similar to a struct.)

The comment in `ParamValue` also describes how we can achieve `None`.